web/views/image.php

```diff
@@ -76,6 +76,22 @@ function imagescale($image, $new_width, $new_height = -1, $mode = 0) {
       return;
     }
 
+    # if alarm, get the fid of the first alarmed frame if available and let the
+    # fid= code continue processing it
+    if ( $_REQUEST['fid'] == 'alarm' ) {
+      $Frame = Frame::find_one(array('EventId'=>$_REQUEST['eid'], 'Type'=>'Alarm'));
```

Comment:
Still have the problem that because we aren't doing any ordering, we could get the last, the first or any alarm frame in between.

Reply:
oh I thought it would always be ordered because of their occurrence in sequence?

Reply:
Shouhld be $Frame = Frame::find_one(array('EventId'=>$_REQUEST['eid'], 'Type'=>'Alarm'),array('order'=>'Id DESC'));